plugins/inputs/mongodb/mongodb.go

```diff
@@ -144,7 +144,17 @@ func (m *MongoDB) Start(telegraf.Accumulator) error {
 	return nil
 }
 
-func (m *MongoDB) Stop() {}
+// Stop disconnect mongo connections when stop or reload
+func (m *MongoDB) Stop() {
+	for _, server := range m.clients {
+		// prefer not to defer inside loops
+		func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			_ = server.client.Disconnect(ctx)
+		}()
```

Comment:
Hi - any reason we cannot just do the following?

```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
		_ = server.client.Disconnect(ctx)
		cancel()
```

Reply:
> Hi - any reason we cannot just do the following?

using defer can ensure that the cancel method can been executed even if the following code raise panic.

Reply:
Yeah, if we panic we've already lost :)

Reply:
got it